packages/bpk-component-autosuggest/src/BpkAutosuggestSuggestion.js

```diff
@@ -16,15 +16,26 @@
  * limitations under the License.
  */
 
+/* @flow strict */
+
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { type Node } from 'react';
 import { cssModules } from 'bpk-react-utils';
 
 import STYLES from './BpkAutosuggest.scss';
 
 const getClassName = cssModules(STYLES);
 
-const BpkSuggestion = props => {
+type Props = {
+  value: Node,
+  indent: boolean,
+  className: ?string,
+  icon: Function,
```

Comment:
I'm wondering if this should be a `Node` if it's a React element that can be passed in. `Function` is technically correct, as React components are technically functions, but it feels like it could be too broad.

Reply:
Will have a look at this, took this from another example in Backpack codebase

Reply:
I chose this as the propTypes at the bottom says its type is `func` so matched the flow type to it, should it still be changed?

Reply:
Ah ok. In this case I would still use a function but make it more specific, like `() => mixed`

Reply:
I probably try using `Element` instead of `() => mixed` as that is too generic whereas we want to be more specific here to restrict to the correct expectation which is what we have done in other places that do this

Reply:
So had a play around with different types here and Function is the best we can use here given the prop is declared as a func in `propTypes` as setting to `() => mixed` or `Element` throws errors that `null` is not a valid function see attached
![Screenshot 2020-08-05 at 10 37 25](https://user-images.githubusercontent.com/8831547/89397739-21807b80-d708-11ea-9e89-bfe724362848.png)

![Screenshot 2020-08-05 at 10 36 18](https://user-images.githubusercontent.com/8831547/89397729-1e858b00-d708-11ea-84cd-4d207611b4d4.png)